src/main/java/com/datadog/ServerlessAzureAgent.java

```diff
@@ -12,16 +12,40 @@
 
 public class ServerlessAzureAgent {
     private static final Logger log = LoggerFactory.getLogger(ServerlessAzureAgent.class);
-    private static final String fileName = "datadog-serverless-trace-mini-agent";
-    private static final String tempDirPath = "/tmp/datadog";
+    private static final String os = System.getProperty("os.name").toLowerCase();
+
+    public static boolean isWindows() {
+        return os.contains("win");
+    }
+
+    public static boolean isLinux() {
+        return os.contains("linux");
+    }
 
     public static void premain(String agentArgs, Instrumentation instrumentation) {
+        final String fileName;
+        final String tempDirPath;
+
+        if (isWindows()) {
```

Comment:
Nit: switch statements are good for this type of logic and might be cleaner to read for some. You would also be able to remove the isWindows() and isLinux() code if you can just check the OS. Not sure if you use them elsewhere though.